### PR TITLE
feat(user): Auth and user endpoints complete

### PR DESCRIPTION
**Pull request by codyleyhan**
# Description
- Adjusts google oAuth to just accept and verify token now
    - once verified issues a JWT access token
- Fixes the users spelling error #2 
- Adds user endpoints for
1. /Login
2. /users/ with pagination
3. /users/:id
4. /users/@me
- adds in user claims 
- adds in a user auth middleware
- updates the logger to reduce code duplication

# Steps to reproduce
- Ensure that you have a fresh DB
- Run the server
- Use the front-end to get your oauth token
- post this token to /api/v1/login (now logged in and will return JWT access token)
- can now hit /users/@me to see your user info
- For /users/ adjust in DB to give yourself auth_level = 1 (admin level) 
- can also hit /users/:id unauthenticated

# Tested
- added unit tests for the Service layer, which handles all the usecases

# Approved
- [x] Jahan
- [x] Luke

### CODE REVIEW

File: adapters/http/user.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	return &UserController{
+		logger:     l,
+		service:    u,
+		authCookie: a,
+	}
+}
+
+// MountRoutes mounts the auth routes
+func (u *UserController) MountRoutes(c *echo.Group) {
+	c.GET("/users/", u.list, auth.NewAuthMiddleware(services.UserLevel, u.logger))
```
Comment (epicchewy) [inline adapters/http/user.go:53]: extra backslash at end?

File: services/user.go [hunk 2]
```diff
@@ -1,29 +1,104 @@
...
+
+// Login checks the token for a valid login, stores user info, and generates an auth token
+func (u *UserService) Login(token string) (string, error) {
+	googleUser, err := u.authorizer.UserLogin(token)
+
+	if err != nil {
+		return "", err
+	}
+	var user *models.User
+	user, err = u.store.GetByEmail(googleUser.Email)
```
Comment (epicchewy) [inline services/user.go]: assign user here?

File: utils/auth/google_authorizer_test.go [hunk 3]
```diff
@@ -1,32 +1,74 @@
...
+	user, err := authorizer.UserLogin(token)
 
-	authorizer.NewUserLogin(c)
+	if err != nil {
+		t.Errorf("With valid response, there should have been no error, got %s", err.Error())
+	}
 
-	if res.Code != http.StatusFound {
-		t.Errorf("expected status code to be 302 found, instead %d", res.Code)
+	if !reflect.DeepEqual(expectedUser, *user) {
```
Comment (epicchewy) [inline utils/auth/google_authorizer_test.go:58]: move to asserts eventually?

File: services/user.go [hunk 4]
```diff
@@ -1,29 +1,104 @@
...
+	if err != nil && err != postgres.ErrNoUserFound {
+		return "", err
+	}
+
+	if user == nil {
```
Comment (epicchewy) [inline services/user.go]: leave comment about adding new user because we're using google auth for all users

**Review by epicchewy — CHANGES_REQUESTED** [review]
minor changes for clarity

File: adapters/http/user.go (hunk 1 above)
Comment (codyleyhan) [reply adapters/http/user.go:53]: Yeah needed

**Comment by codyleyhan** [comment]
Fat 1000 line PR coming in hot

**Review by epicchewy — APPROVED** [review]

**Comment by epicchewy** [comment]
lgtm

**Review by jcherianucla — APPROVED** [review]
Other than the limit, everything looks good to me

**Pull request merged**